deploy/roles/webserver/tasks/main.yml

```diff
@@ -0,0 +1,171 @@
+---
+- name: Install required packages
+  sudo: true
+  apt: pkg={{ item }} state=installed
+  with_items:
+    - build-essential
+    - curl
+    - git
+    - git-core
+    - libcurl4-openssl-dev
+    - libgmp3-dev
+    - libpq-dev
+    - libreadline-dev
+    - libssl-dev
+    - libxml2-dev
+    - libxslt1-dev
+    - libyaml-dev
+    - nginx-extras
+    - zlib1g-dev
+    - nodejs
+
+  # LetsEncrypt setup, these must happen in order and should only happen on
+  # very first deploy
+- name: Check if SSL cert has been installed
+  shell: "test ! -e {{ ssl_certificate }}"
+  register: obtain_letsencrypt_cert
```

Comment:
This is an elegant solution, wasn't aware this was possible with Ansible. Would it make sense to move all SSL tasks into their own role?